packages/liquidator/index.js

```diff
@@ -132,26 +131,37 @@ async function run({
       pollingDelay,
       errorRetries,
       errorRetriesTimeout,
-      priceFeedConfig: customPricefeedConfig,
+      priceFeedConfig,
       liquidatorConfig,
       liquidatorOverridePrice
     });
 
     // Load unlocked web3 accounts, get the networkId and set up price feed.
-    const [priceFeed] = await Promise.all([
-      createReferencePriceFeedForEmp(logger, web3, new Networker(logger), getTime, empAddress, customPricefeedConfig)
-    ]);
+    const priceFeed = await createReferencePriceFeedForEmp(
+      logger,
+      web3,
+      new Networker(logger),
+      getTime,
+      empAddress,
+      priceFeedConfig
+    );
+
     if (!priceFeed) {
       throw new Error("Price feed config is invalid");
     }
-    logger.debug({
-      at: "Liquidator#index",
-      message: `Using an ${customPricefeedConfig.decimals} decimal price feed`
-    });
 
     // Create the ExpiringMultiPartyClient to query on-chain information, GasEstimator to get latest gas prices and an
     // instance of Liquidator to preform liquidations.
-    const empClient = new ExpiringMultiPartyClient(logger, getAbi("ExpiringMultiParty"), web3, empAddress);
+    const empClient = new ExpiringMultiPartyClient(
+      logger,
+      getAbi("ExpiringMultiParty"),
+      web3,
+      empAddress,
+      collateralTokenDecimals,
+      syntheticTokenDecimals,
+      priceFeed.getPriceFeedDecimals()
```

Comment:
So currently we hardcode that every type of `PriceFeed` returns 18 for `getPriceFeedDecimals`, but imagine that we want to change this to 8 (for some reason) for one EMP.

Wouldn't it be easier to make this change if set the decimals to 8 within this `index.js` entrypoint, and simultaneously passed it to the `PriceFeed` and the `ExpiringMultiPartyClient`?

My main point is that if we want to change the `PriceFeed`'s decimals to something other than 18, then how can we dynamically change it for any EMP config.

Reply:
So the way this works is that the price feed itself contains a `priceFeedDecimals` that defines how to scale the price feed. this can be parameterized within the json or is added to the default config by re-using the method from common.

This is auto-detected based on the identifier or is parameterizable by the user.

i.e this _is_ set within the `index.js` file entry point and is fed downstream to all associated modules and clients.

Does that make sense?

Reply:
Yep makes sense